src/i2c.rs

```diff
@@ -97,7 +109,7 @@ mod tests {
         let expectations = [Transaction::write(0x58, vec![0xab, 0xcd])];
         let mut mock = I2cMock::new(&expectations);
 
-        i2c::write_command(&mut mock, 0x58, 0xabcd).unwrap();
+        i2c::write_command(&mut mock, 0x58, 0xabcd as u16).unwrap();
```

Comment:
I don't like that now casts are necessary. So the advantage claimed in the PR "It enables type inference, so you don't need to manually specify the type when calling the function." doesn't hold, since we now need to specify the type.

Reply:
All 100% valid points I was hoping you wouldn't bring up! Admittedly these changes just make it more ergonomic for me and so I thought I'd contribute them back as an option.

You're right the address stuff is more about future proofing than anything necessary. I'm not familiar with all of Sensirion's products and future plans so I thought it just might be a nice-to-have.

As for the `Into<u16>` that is a fair assessment but I'd say it would be possible to split the difference and use `TryInto<u16>`...but then we'd have to figure out what type of error to throw if the conversion fails. But if that isn't of interest I understand and will just keep a separate branch I guess.

Reply:
Ok I pushed those changes. Actually it feels a lot stronger of an argument now. Now your API shares a common error for the Result type across the API and it increases ergonomics since it supports both ways of passing in commands.